airbyte-integrations/connectors/source-gitlab/source_gitlab/manifest.yaml

```diff
@@ -353,6 +360,55 @@ definitions:
     $parameters:
       path: "projects/{{ stream_slice.id }}/labels"
 
+  project_dora_metrics_deployment_frequency_stream:
+    name: "project_dora_metrics_deployment_frequency"
+    $ref: "#/definitions/base_projects_child_stream"
+    primary_key: ["project_id"]
```

Comment:
Wouldn't that rather be ["project_id", "date"]?

Reply:
I couldn't see where to have this "date" field : https://docs.gitlab.com/ee/api/dora/metrics.html

Reply:
Any comment @Brzhk ?

Reply:
It's part of the data in the response, and could be added to the primary key to make each record identifiable.

```json
[
  { "date": "2021-03-01", "value": 3 },
  { "date": "2021-03-02", "value": 6 },
  { "date": "2021-03-03", "value": 0 },
  { "date": "2021-03-04", "value": 0 },
  { "date": "2021-03-05", "value": 0 },
  { "date": "2021-03-06", "value": 0 },
  { "date": "2021-03-07", "value": 0 },
  { "date": "2021-03-08", "value": 4 }
]
```

then `primary_key: ["project_id", "date"]`?

Reply:
This would make the primary key unique per record and let us deduplicate records reliably. I'm not yet confident in my comprehension of all the sync mechanics, and concepts like cursor fields to go any further though.